hackathon_site/dashboard/frontend/src/slices/hardware/cartSlice.ts

```diff
@@ -1,4 +1,62 @@
+import {
+    createEntityAdapter,
+    createSelector,
+    createSlice,
+    PayloadAction,
+} from "@reduxjs/toolkit";
+import { RootState } from "slices/store";
+
 export interface CartItem {
```

Comment:
minor: Could you put this in types.ts? Just because we're putting all the general interfaces there and interfaces that are only specific to the slice are here.

Reply:
@YuYing-Liang Do you mean to move these imports to types.ts?
`import {
    createEntityAdapter,
    createSelector,
    createSlice,
    PayloadAction,
} from "@reduxjs/toolkit";
import { RootState } from "slices/store";`

Reply:
Ah no, the interface for CartItem

Reply:
@YuYing-Liang do you also want me to move the CartExtraState interface as well?

Reply:
The `CartExtraState` is specific to this file, we don't use it in other places so you can leave it in here

Reply:
I put `CartItem` in here originally because it isn't actually an API type, but having it in there works just fine 👌